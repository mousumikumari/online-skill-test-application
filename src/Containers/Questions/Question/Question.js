/* eslint-disable react/destructuring-assignment */
import React from "react";
import PropTypes from "prop-types";
import { Button } from "react-bootstrap";

const Question = (props) => {
  const {
    questionCount, questionId, submitTest, questionText, onNext, onPrevious, options,
  } = props;
  let second = props.quizTime % 60;
  let minute = Math.floor(props.quizTime / 60);
  minute = minute.toString().length === 1 ? `0${minute}` : minute;
  second = second.toString().length === 1 ? `0${second}` : second;

  if (second === "00" && minute === "00") {
    submitTest();
  }
  return (
    <div className="container">
      <div className="Countdown-time">
        Timer : {minute} : {second}
      </div>
      <div className="questions">
        <h4 id="question">
          Question {questionId + 1} : {questionText}
        </h4>
        <div id="">
          <div className="radio">
            {options.map((choice) => (
              <div key={choice}>
                <label className="radio-inline" htmlFor="{choice}">
                  <input
                    type="checkbox"
                    value={choice}
                    key={choice}
                    checked={
                      props.answer.length > 0
                        ? props.answer.includes(choice)
                        : false
                    }
                    onChange={() => props.onChange(choice)}
                  />
                  {choice}
                </label>
                <br />
              </div>
            ))}
          </div>
        </div>

        {questionId !== 0 && (
          <Button
            variant="info"
            onClick={onPrevious}
            className="questionaireButton"
          >
            Previous
          </Button>
        )}
        {questionId < questionCount - 1 && (
          <Button
            variant="info"
            className="questionaireButton"
            onClick={onNext}
          >
            Next
          </Button>
        )}
        <Button
          variant="info"
          id="submitTest"
          onClick={submitTest}
          className="questionaireButton"
        >
          Submit Test
        </Button>
      </div>
    </div>
  );
};

Question.propTypes = {
  quizTime: PropTypes.number.isRequired,
  questionCount: PropTypes.number.isRequired,
  questionId: PropTypes.number.isRequired,
  submitTest: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
  onPrevious: PropTypes.func.isRequired,
  answer: PropTypes.func.isRequired,
  questionText: PropTypes.string.isRequired,
  onNext: PropTypes.func.isRequired,
  options: PropTypes.func.isRequired,
};

export default Question;
