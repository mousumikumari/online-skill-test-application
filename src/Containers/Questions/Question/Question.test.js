/* eslint-disable */
import React from "react";
import renderer from "react-test-renderer";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import Question from "./Question";

const historyMock = { push: jest.fn() };

configure({ adapter: new Adapter() });
test("Question Snapshot test", () => {
  const component = renderer.create(
    <Question
    questionId = "1"
    questionCount= "10"
    onNext = { historyMock }
    onPrevious={ historyMock }
    onChange={ historyMock }
    submitTest={ historyMock }
    answer=""
    clearResponse={ historyMock }
    isQuizSubmitted = "false"
    quizTime= "600"
    options = {[]}
    questionId= "0"
    questionText="What is ReactJs in MVC?"
    />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
