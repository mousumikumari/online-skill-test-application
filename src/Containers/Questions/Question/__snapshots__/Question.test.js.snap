// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question Snapshot test 1`] = `
<div
  className="container"
>
  <div
    className="Countdown-time"
  >
    Timer : 
    10
     : 
    00
  </div>
  <div
    className="questions"
  >
    <h4
      id="question"
    >
      Question 
      01
       : 
      What is ReactJs in MVC?
    </h4>
    <div
      id=""
    >
      <div
        className="radio"
      />
    </div>
    <button
      className="questionaireButton btn btn-info"
      disabled={false}
      onClick={
        Object {
          "push": [Function],
        }
      }
      type="button"
    >
      Previous
    </button>
    <button
      className="questionaireButton btn btn-info"
      disabled={false}
      onClick={
        Object {
          "push": [Function],
        }
      }
      type="button"
    >
      Next
    </button>
    <button
      className="questionaireButton btn btn-info"
      disabled={false}
      id="submitTest"
      onClick={
        Object {
          "push": [Function],
        }
      }
      type="button"
    >
      Submit Test
    </button>
  </div>
</div>
`;
